pymeasure/instruments/deltaelektronika/sm7045d.py

```diff
@@ -0,0 +1,125 @@
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_range
+
+from time import sleep
+import numpy as np
+
+
+class SM7045D(Instrument):
+    """ This is the class for the SM 70-45 D power supply.
+
+    .. code-block:: python
+
+        source = SM7045D("GPIB::8")
+
+        source.ramp_to_zero(1)               # Set output to 0 before enabling
+        source.enable()                      # Enables the output
+        source.current = 1                   # Sets a current of 1 Amps
+
+    """
+    VOLTAGE_RANGE = [0, 70]
+    CURRENT_RANGE = [0, 45]
+
+    voltage = Instrument.control(
+        "SO:VO?", "SO:VO %g",
+        """ A floating point property that represents the output voltage
+        setting of the power supply in Volts. This property can be set. """,
+        validator=strict_range,
+        values=VOLTAGE_RANGE
+    )
+
+    current = Instrument.control(
+        "SO:CU?", "SO:CU %g",
+        """ A floating point property that represents the output current of
+        the power supply in Amps. This property can be set. """,
+        validator=strict_range,
+        values=CURRENT_RANGE
+    )
+
+    max_voltage = Instrument.control(
+        "SO:VO:MA?", "SO:VO:MA %g",
+        """ A floating point property that represents the maximum output
+        voltage of the power supply in Volts. This property can be set. """,
+        validator=strict_range,
+        values=VOLTAGE_RANGE
+    )
+
+    max_current = Instrument.control(
+        "SO:CU:MA?", "SO:VO:MA %g",
+        """ A floating point property that represents the maximum output
+        current of the power supply in Amps. This property can be set. """,
+        validator=strict_range,
+        values=CURRENT_RANGE
+    )
+
+    measure_voltage = Instrument.measurement(
+        "ME:VO?",
+        """ Measures the actual output voltage of the power supply in
+        Volts. """,
+    )
+
+    measure_current = Instrument.measurement(
+        "ME:CU?",
+        """ Measures the actual output current of the power supply in
+        Amps. """,
+    )
+
+    rsd = Instrument.measurement(
+        "SO:FU:RSD?",
+        """ Check whether remote shutdown is enabled/disabled and thus if the
+        output of the power supply is disabled/enabled. """,
+    )
+
+    def __init__(self, resourceName, **kwargs):
+        super(SM7045D, self).__init__(
+            resourceName,
+            "Delta Elektronika SM 70-45 D",
+            **kwargs
+        )
+
+    def enable(self):
+        # Disable remote shutdown, hence output will be enabled.
+        self.write("SO:FU:RSD 0")
+
+    def disable(self):
+        # Enables remote shutdown, hence input will be disabled.
+        self.write("SO:FU:RSD 1")
+
+    def ramp_to_current(self, target_current, current_step):
```

Comment:
I think it is worth setting a default for `current_step`.

Reply:
That's a good suggestion, i implemented it